### ISSUE TITLE
User registration with groups inserts odd rows

### ISSUE THREAD

**Issue by Swader** [psecio/gatekeeper#60]
When I put this POST data directly into the `Gatekeeper::register` method:

<img width="278" alt="screenshot 2016-03-13 09 12 03" src="https://cloud.githubusercontent.com/assets/1430603/13727667/d9befad2-e8fb-11e5-8837-8f6368c9f81f.png">

Then my records go from this:

<img width="1213" alt="screenshot 2016-03-13 09 08 27" src="https://cloud.githubusercontent.com/assets/1430603/13727669/e22ee20e-e8fb-11e5-9cfb-059ead261ef0.png">

to this:

<img width="1213" alt="screenshot 2016-03-13 09 10 04" src="https://cloud.githubusercontent.com/assets/1430603/13727670/e764e2fa-e8fb-11e5-9849-2aebc6ad2552.png">

and I also get an additional MySQL error: `DB ERROR: [23000] Duplicate entry '1-12' for key 'group_id'`

In other words, when inserting this data, GK not only tries to insert a group-user pair twice, it also for some reason grabs the ID of the last inserted user-group pair row and uses that ID as the next user ID.

Am investigating further.

**Comment by Swader** [comment]
Note that things work just fine if I first save the user, then iterate through the group IDs and use them as `addGroup` on the user instance.
